### PR TITLE
Pl refactor v0.1.13

### PR DESCRIPTION
**Pull request by paulflang**
This will need SBML#master, which is several commits ahead of v0.10.1

### CODE REVIEW

**Comment by paulflang** [comment]
src/reactionsystems.jl has become a rather big file. I am planning to split it up in multiple files in a future PR.

File: test/misc.jl [hunk 1]
```diff
@@ -0,0 +1,178 @@
+const case_ids = [7, 22,
+                #140, requires structural_simplify fix
+                170, 325,
+                # 679 requires structural_simplify fix
+                ]
+
+# const case_ids = [325]
+const cases = map(x -> x[end-4:end], .*("0000", string.(case_ids)))
+
+const algomap = Dict("00177" => Rodas4(),
```
Comment (anandijain) [inline test/misc.jl:11]: misc is not a great name for this file. what are these testing?

Comment (paulflang) [reply test/misc.jl:11]: It is just a few representative test suite cases that fail, despite the tests in the other files included in runtests.jl pass.

I mean in reactionsystem.jl there are largely unit tests, then there are some that are more integration tests. Perhaps misc.jl could be renamed to functional_tests.jl. What do you think?

Comment (anandijain) [reply test/misc.jl:11]: misc is okay, but could you add comments for what each of these cases is testing

**Comment by anandijain** [comment]
otherwise LGTM but make sure to squash and merge

**Comment by paulflang** [comment]
Thanks! Squash and merge messes with history, though.

**Comment by anandijain** [comment]
there is only 109 commits total, this PR has 81. do you want this patch revision to account for more almost half of the entire repository? i dont

**Comment by paulflang** [comment]
@ChrisRackauckas what is the philosophy regarding clean vs accurate commit history (squash merge vs merge) in SciML? All I see in the ColPracs on "squash" is

> You should not squash down commits while review is still on-going.
> Squashing commits prevents the reviewer being able to see what commits are added since the last review.

File: src/reactionsystem.jl [hunk 2]
```diff
@@ -1,192 +1,165 @@
...
-function from_noncombinatoric(rl::Num, stoich::Union{Vector{<:Real},Nothing}, only_use_rate::Bool)
-    if !isnothing(stoich) && !only_use_rate
-        coef = 1
-        for s in stoich
-            isone(s) && continue
-            coef *= factorial(s)
+""" Infer forward and reverse components of bidirectional kineticLaw """
+function get_unidirectional_components(bidirectional_math)
+    err = "Cannot separate bidirectional kineticLaw `$bidirectional_math` to forward and reverse part. Please make reaction irreversible or rearrange kineticLaw to the form `term1 - term2`."
+    bmdict = bidirectional_math.val.dict
```
Comment (anandijain) [inline src/reactionsystem.jl]: I don't think this is how you are supposed to use the term interface.

I just hit `ERROR: type Div has no field dict`. I think you need to check `operation(Symbolics.value(term))` to check if its `Add` or something

Comment (paulflang) [reply src/reactionsystem.jl]: Oh, looks like I need to revert [c83ee5d](https://github.com/SciML/SBMLToolkit.jl/pull/71/commits/c83ee5d0cb9588ee74f75f6d90c4992224c6da2b)

Comment (paulflang) [reply src/reactionsystem.jl]: reason for [c83ee5d](https://github.com/SciML/SBMLToolkit.jl/pull/71/commits/c83ee5d0cb9588ee74f75f6d90c4992224c6da2b) in the first place, was to enable a bidirectional rate law in the QSP model we are working on rn. But splitting bidirectional rate laws is a pain. So perhaps just do this particular case manually in the SBML.

**Comment by paulflang** [comment]
Decided to merge without squashing, in case sth comes up that we need to revert a commit again, like 2 days ago.

**Pull request merged**